### PR TITLE
(Dropbox) Working package with update script

### PR DESCRIPTION
**Pull request by The-Running-Dev**

### CODE REVIEW

**Comment by rlabrecque** [comment]
Can you see what's up with the dropboxo branch? cc: @majkinetor 

And, can you update the package to 23.4.18?

**Comment by The-Running-Dev** [comment]
The "dropboxo" branch is not mine, I don't  know what is up with it.

I can update this to 23,4,16, but isn't that the point of the automatic updater?

File: automatic/dropbox/dropbox.nuspec [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+  <metadata>
+    <id>dropbox</id>
+    <version>0.0</version>
+    <title>Dropbox</title>
+    <authors>Dropbox</authors>
+    <owners>chocolatey,ferventcoder,the-running-dev</owners>
+    <licenseUrl>http://www.dropbox.com/terms</licenseUrl>
+    <projectUrl>https://www.dropbox.com/</projectUrl>
+    <iconUrl>https://raw.githubusercontent.com/chocolatey/chocolatey-coreteampackages/master/icons/dropbox.svg</iconUrl>
```
Comment (AdmiringWorm) [inline automatic/dropbox/dropbox.nuspec]: this should actually use the rawgit cdn, not the raw github user content.
But this should be handled with an existing script during the build, so no need to change anything here.

File: automatic/dropbox/dropbox.nuspec [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    <description>
+    Dropbox is a free service that lets you bring all your photos, docs,
+    and videos anywhere. This means that any file you save to your
+    Dropbox will automatically save to all your computers,
+    phones and even the Dropbox website. Dropbox also makes it super easy
+    to share with others, whether you're a student or professional,
+    parent or grandparent. Even if you accidentally spill a latte on your
+    laptop, have no fear! You can relax knowing that Dropbox always has
+    you covered, and none of your stuff will ever be lost.
+    </description>
```
Comment (AdmiringWorm) [inline automatic/dropbox/dropbox.nuspec]: Please add a `## Features` section to the description.

File: automatic/dropbox/dropbox.nuspec [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    <summary>Dropbox is a free service that lets you bring all your photos, docs, and videos anywhere.</summary>
+    <tags>dropbox virtual files file admin freeware cross-platform</tags>
+    <packageSourceUrl>https://github.com/chocolatey/chocolatey-coreteampackages</packageSourceUrl>
```
Comment (AdmiringWorm) [inline automatic/dropbox/dropbox.nuspec]: the package source should be in the format `https://github.com/chocolatey/chocolatey-coreteampackages/tree/master/automatic/{packageName}`

File: automatic/dropbox/tools/chocolateyuninstall.ps1 [hunk 4]
```diff
@@ -0,0 +1,21 @@
+$packageName = 'dropbox'
+$fileType = 'exe'
+$silentArgs = '/S'
+$uninstallerPath = (getDropboxRegProps).UninstallString
+
+if ($uninstallerPath) {
+    Uninstall-ChocolateyPackage $packageName $fileType $silentArgs $uninstallerPath
+}
+
+function getDropboxRegProps() {
```
Comment (AdmiringWorm) [inline automatic/dropbox/tools/chocolateyuninstall.ps1]: this should be in a seperate `helpers.ps1` file, as per [1.1.8 Clean code](https://github.com/chocolatey/chocolatey-coreteampackages/wiki/Contributing-guidelines#118-clean-code)

File: automatic/dropbox/dropbox.nuspec [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  </metadata>
+  <files>
+    <file src="tools\**" target="tools" />
+  </files>
+</package>
```
Comment (AdmiringWorm) [inline automatic/dropbox/dropbox.nuspec]: you're also missing a few urls that should be provided:
`mailingListUrl`: https://www.dropboxforum.com/
`docsUrl`: https://www.dropbox.com/help (perhaps, not completely sure)

File: automatic/dropbox/dropbox.nuspec [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+  <metadata>
+    <id>dropbox</id>
+    <version>0.0</version>
+    <title>Dropbox</title>
+    <authors>Dropbox</authors>
+    <owners>chocolatey,ferventcoder,the-running-dev</owners>
+    <licenseUrl>http://www.dropbox.com/terms</licenseUrl>
```
Comment (AdmiringWorm) [inline automatic/dropbox/dropbox.nuspec]: use the https version of this url

File: automatic/dropbox/dropbox.nuspec [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    <projectUrl>https://www.dropbox.com/</projectUrl>
+    <iconUrl>https://raw.githubusercontent.com/chocolatey/chocolatey-coreteampackages/master/icons/dropbox.svg</iconUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
```
Comment (AdmiringWorm) [inline automatic/dropbox/dropbox.nuspec]: I think this should be set to `true`, from skipping the dropbox terms it looks like it.

**Comment by The-Running-Dev** [comment]
I've made the requested changes, please review.

File: automatic/dropbox/dropbox.nuspec [hunk 8]
```diff
@@ -6,11 +6,14 @@
...
+    <licenseUrl>https://www.dropbox.com/terms</licenseUrl>
     <projectUrl>https://www.dropbox.com/</projectUrl>
-    <iconUrl>https://raw.githubusercontent.com/chocolatey/chocolatey-coreteampackages/master/icons/dropbox.svg</iconUrl>
-    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <iconUrl>https://cdn.rawgit.com/chocolatey/chocolatey-coreteampackages/9d90583a4103d4a9db86425a5fbda79cc845b4ef/icons/dropbox.svg</iconUrl>
+    <mailingListUrl>https://www.dropboxforum.com/</mailingListUrl>
+    <docsUrl>https://www.dropbox.com/help</docsUrl>
+    <requireLicenseAcceptance>true</requireLicenseAcceptance>
     <description>
+    ## Features
```
Comment (AdmiringWorm) [inline automatic/dropbox/dropbox.nuspec]: Sorry, that's not I meant when I said to add a feature section.
I meant you should add a features section after the current description.
Like what is done with the [qbittorrent](https://chocolatey.org/packages/qbittorrent) package (and most other packages in this repo)

File: automatic/dropbox/dropbox.nuspec [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    - File requests
+    - Smart Sync
+    - Team folder
+    - Admin console
+    - Centralized billing
+    - Company-managed groups
+    - Unlimited API access to platform partners
+    - API access for data transport
+    - Priority email support
+    - Live chat support
```
Comment (AdmiringWorm) [inline automatic/dropbox/dropbox.nuspec]: wow, that was a larger features list than I expected.
Nice work.

**Comment by AdmiringWorm** [comment]
@The-Running-Dev have you opened a PR to @ferventcoder's repo removing this package?
when you have and it have been merged, please post a message when it has here.

**Comment by The-Running-Dev** [comment]
Done. I'm waiting on him to merge it. https://github.com/ferventcoder/chocolatey-packages/pull/243

**Comment by AdmiringWorm** [comment]
@The-Running-Dev 👍 
btw, could you rebase this PR to the latest version of the master branch?
I would like to have the appveyor build run available for this PR.
(`git rebase`, not `git merge`)

**Comment by The-Running-Dev** [comment]
I don't usually rebase, I tried, not sure if I did it right.

**Comment by AdmiringWorm** [comment]
@The-Running-Dev unfortunately the rebasing you did wasn't correct.
you first need to update your local master branch with the changes from this repo.
then switch over to the `feat-Dropbox` branch and run `git rebase master`

**Comment by AdmiringWorm** [comment]
however I can do the rebasing if you want to, or are unable to get it work correctly.

**Comment by The-Running-Dev** [comment]
How about now? I really need more practice with rebasing, I'll work on that.

**Comment by AdmiringWorm** [comment]
@The-Running-Dev if you look at the commit history for this PR you'll notice something have gone wrong.
My guess is that your local master branch differs from the original master branch here in the repo.
Try resetting your local branch to the original master banch, and try rebasing it again (you'll definitely get merge conflicts when trying to rebase this PR).

**Comment by AdmiringWorm** [comment]
@The-Running-Dev I rebased your PR now, also added a new commit

**Comment by The-Running-Dev** [comment]
Thanks for that. I'm sorry for the delay, I did mix it up somehow but haven't had time to work on any Chocolatey stuff in the last few days. Is this all set to merge now?

**Pull request merged**